shared/profile/edit-profile/index.js

```diff
@@ -0,0 +1,79 @@
+/* @flow */
+import React, {Component} from 'react'
+import {connect} from 'react-redux'
+import {editProfile} from '../../actions/profile'
+import {maxProfileBioChars} from '../../constants/profile'
+
+import Render from './render'
+import type {Props} from './render'
+import {navigateUp} from '../../actions/router'
+
+type State = {
+  bio: string,
+  fullname: string,
+  location: string
+}
+
+class EditProfile extends Component<void, Props, State> {
+  props: Props;
+  state: State;
+
+  static parseRoute (currentPath, uri) {
```

Comment:
https://github.com/keybase/client/blob/master/shared/router/meta-navigator.js#L93

It expects the subroute to have a parseRoute, I was wrong when I said it didn't.

It uses it to get info for componentAtTop like the title, and optionally component.

In the code it gets the nextComponent correct from the parent in the route, but if this didn't have a parseRoute it wouldn't run the loop again and the component that would be renderered would be the parent and not the child in nextComponent.